src/Components/GrantsFinalizeShow.js

```diff
@@ -1,414 +1,143 @@
-    newSuccessful,
-    newPurpose,
-  }) => {
-    updateGrant(organizationClient, id, {
-      title: newTitle,
-      rfp_url: newRfpUrl,
-      deadline: newDeadline,
-      submitted: newSubmitted,
-      successful: newSuccessful,
-      purpose: newPurpose,
-      sections: [],
-      organization_id: organizationId,
-      funding_org_id: fundingOrgId,
-    })
-      .then((grant) => {
-        toggleHidden();
-        handleClose();
-        setTitle(grant.title);
-        setRfpUrl(grant.rfp_url);
-        setDeadline(grant.deadline);
-        setSubmitted(grant.submitted);
-        setSuccessful(grant.successful);
-        setPurpose(grant.purpose);
-      })
-      .catch((error) => {
-        console.log("grant update error", error);
-      });
-  };
-
-  const handleCancel = (event) => {
-    handleClose();
-  };
-
-  const handleCopyGrant = ({
-    copyTitle,
-    copyRfpUrl,
-    copyDeadline,
-    copyPurpose,
-  }) => {
-    if (currentOrganizationId) {
-      copyGrant(organizationClient, grant.id, {
-        title: copyTitle,
-        rfp_url: copyRfpUrl,
-        deadline: copyDeadline,
-        purpose: copyPurpose,
-      })
-        .then((grant) => {
-          console.log(grant.id);
-          setCopiedGrantId(grant.id);
-          setShowCopyModal(true);
-          setSuccessfulCopy(true);
-          toggleCopyGrantHidden();
-        })
-        .catch((error) => {
-          console.log("grant copy error", error);
-          setShowCopyModal(true);
-          setSuccessfulCopy(false);
-        });
+  const getGrant = useCallback(() => {
+    if (!organizationClient) {
+      return;
     }
-  };
-
-  const handleSectionDelete = () => {
-    axios
-      .delete(
-        `/api/organizations/${currentOrganizationId}/sections/` +
-          props.section.id,
-        {
-          headers: { Authorization: `Bearer ${localStorage.token}` },
-        }
-      )
-      .then((response) => {
-        console.log(response);
-      })
-      .catch((error) => {
-        console.log(error);
-      });
-  };
-
-  const updateSections = (newSection) => {
-    const sections = sections.map((section) => {
-      if (section.id === newSection.id) {
-        section.title = newSection.title;
-        section.text = newSection.text;
-        section.wordcount = newSection.wordcount;
-      }
-      return section;
+    GrantsService.getGrant(organizationClient, grantId)
+      .then((grant) => setGrant(grant))
+      .catch((error) => setErrors([error]))
+      .finally(() => setLoading(false));
+  }, [organizationClient, grantId]);
+
+  const handleSubmitSectionForm = ({ newSectionFields, precedingSection }) => {
+    createGrantSection(organizationClient, grantId, {
+      title: newSectionFields.title,
+      text: newSectionFields.html,
+      grant_id: grantId,
+      sort_order: precedingSection.sort_order + 1,
+      // TODO: consider moving wordcount to server-side
+      wordcount: countWords(newSectionFields.text),
```

Comment:
Looking more into this, this might not be the best approach, but def something to think about